### ISSUE TITLE
Why do you divide the image tensor by 255.

### ISSUE THREAD

**Issue by dwang68** [sgrvinod/a-PyTorch-Tutorial-to-Image-Captioning#60]
In the CaptionDataset's __get_item__() method, you devided the image tensor by 255.  Is it for regularization or for something else? 

img = torch.FloatTensor(self.imgs[i // self.cpi] / 255.)

Thanks!

**Comment by kmario23** [comment]
Hi @dwang68 ,
  The division by 255 is applied to convert RGB pixel values which are usually in the range `[0, 255]` to `[0.0, 1.0]` range. Please see [`torchvision.transforms.ToTensor`](https://pytorch.org/docs/stable/torchvision/transforms.html#torchvision.transforms.ToTensor) for more information.

**Comment by sgrvinod** [comment]
Hi @dwang68, 

To add to what @kmario23 said, the reason they are in `[0, 255]` is because true color images have 256 values (8-bit) for each channel. It's common to normalize this to `[0, 1]` as the input to a neural network. 

In our case, it is also _necessary_ because we're using a pretrained ResNet from `torchvision` that has been trained on images with values in `[0, 1]` and therefore expects the same format. 

As mentioned [here](https://pytorch.org/docs/stable/torchvision/models.html) -

> The images have to be loaded in to a range of [0, 1] and then normalized using `mean = [0.485, 0.456, 0.406]` and `std = [0.229, 0.224, 0.225]`
